db/models.py

```diff
@@ -535,6 +535,10 @@ class GroupMembership(SoftDeleteModel, table=True):
         default=None,
         sa_column=Column(String(1024), nullable=True)
     )
+    rejection_reason: str | None = Field(
```

Comment:
do we need a separate field for this, or can we just rename the existing field to `reason`?

Reply:
I would keep them seperate until this becomes a problem (we get a lot of rejection reasons)... it makes reconstruction of the DB slightly easier at almost no cost

Reply:
OK, not a big issue either way